synapse/handlers/typing.py

```diff
@@ -253,20 +253,19 @@ async def started_typing(
         self, target_user: UserID, requester: Requester, room_id: str, timeout: int
     ) -> None:
         target_user_id = target_user.to_string()
-        auth_user_id = requester.user.to_string()
 
         if not self.is_mine_id(target_user_id):
             raise SynapseError(400, "User is not hosted on this homeserver")
 
-        if target_user_id != auth_user_id:
+        if target_user != requester.user:
```

Comment:
I'm assuming we can compare two `UserID` directly?

Reply:
yes, thanks to the magic of `attrs` (https://www.attrs.org/en/stable/comparison.html)

Reply:
(I should clarify: the definition of `UserID` is annotated with `@attr.s`, which gives it the above magic. Generally, class instances are only equal if they are the exact same instance or otherwise override `__eq__`)